packages/runtime/container-runtime/src/summaryCollection.ts

```diff
@@ -201,6 +201,9 @@ export class SummaryCollection {
     private readonly pendingSummaries = new Map<number, Summary>();
     private refreshWaitNextAck = new Deferred<void>();
 
+    private lastSummaryTimestamp: number | undefined;
+    private maxAckWaitTime: number | undefined;
+    private pendingAckTimerTimeoutCallback: (() => void) | undefined;
```

Comment:
I was hoping to keep these details out of SummaryCollection if possible. **I think it is totally fine to merge the PR like it is now, since it is simple**, but I was thinking of exposing the time difference more generically.

For example: I imagined something in the watcher that would either resolve the `waitBroadcast()` or `waitAckNack()` early or expose something like `waitTimeSinceOp(ms: number)` that would resolve when that many ms had passed (by op timestamp diff), and then be consumed in RunningSummarizer by `Promise.race([waitBroadcast(), waitTimeSinceOp(x)])`.  And the name of the function could be whatever like `waitOpTimeDiff(...)` or whatever.

Reply:
Isn't the summary watcher watches the summary for its own client?

Reply:
Yeah but we also expose `waitFlushed()` there. So we could have a corresponding function to wait for time since last op. Then also have the ones for the individual summaries.

It can be a later refactor if need be.